src/xhr/src/xhr.ts

```diff
@@ -31,8 +31,23 @@ async function xhr(options: XhrOptions) {
 
     const request = new XhrRequest(`${baseURL}/${resolvedPath}`, xhrOptions)
 
-    // Begin timing how long our request will take
-    request.setTimestamp()
+    /**
+     * Begin timing how long our request will take
+     *
+     * We try to extend window.Request with setTimestamp but it appears to
+     * be disallowed in Safari:
+     * TypeError - request.setTimestamp is not a function
+     *
+     * However, we appear to be able to arbitrarily set and get props in Safari:
+     *   window.Request.timestamp = 1234
+     *   window.Request.timestamp // outputs 1234
+     */
+
+    if (request.setTimestamp) {
```

Comment:
Fixes Safari XHR request.setTimeout undefined error

Reply:
Ok, I pulled this into nora and ran on safari no more `setTimestamp` bug:

![image](https://user-images.githubusercontent.com/142403/73023420-5702ee80-3de0-11ea-82de-3c822e7b9d6f.png)

Reply:
woot woot!